core/adapters/random.go

```diff
@@ -56,33 +69,76 @@ func (ra *Random) TaskType() models.TaskType {
 
 // Perform returns the the proof for the VRF output given seed, or an error.
 func (ra *Random) Perform(input models.RunInput, store *store.Store) models.RunOutput {
-	key, err := getKey(ra, input)
+	key, preSeed, blockHash, blockNum, err := getInputs(ra, input, store)
 	if err != nil {
-		return models.NewRunOutputError(errors.Wrapf(err, "bad key for vrf task"))
-	}
-	seed, err := getSeed(input)
-	if err != nil {
-		return models.NewRunOutputError(errors.Wrap(err, "bad seed for vrf task"))
+		return models.NewRunOutputError(err)
 	}
-	solidityProof, err := store.VRFKeyStore.GenerateProof(key, seed)
+	solidityProof, err := store.VRFKeyStore.GenerateProof(key, preSeed,
+		*blockHash, *blockNum)
 	if err != nil {
 		return models.NewRunOutputError(err)
 	}
 	ethereumByteArray := fmt.Sprintf("0x%x", utils.EVMEncodeBytes(solidityProof[:]))
 	return models.NewRunOutputCompleteWithResult(ethereumByteArray)
 }
 
-// getSeed returns the numeric seed for the vrf task, or an error
-func getSeed(input models.RunInput) (*big.Int, error) {
+// getInputs parses the JSON input for the values needed by the random adapter,
+// or returns an error.
+func getInputs(ra *Random, input models.RunInput, store *store.Store) (
+	key *vrfkey.PublicKey, preSeed *vrf.Seed, blockHash *common.Hash,
+	blockNum *uint64, err error) {
+	key, err = getKey(ra, input)
+	if err != nil {
+		return nil, nil, nil, nil, errors.Wrapf(err, "bad key for vrf task")
+	}
+	preSeed, err = getPreSeed(input)
+	if err != nil {
+		return nil, nil, nil, nil, errors.Wrap(err, "bad seed for vrf task")
+	}
+	blockHash, blockNum, err = getBlockData(input)
+	if err != nil {
+		return nil, nil, nil, nil, err
+	}
+	return key, preSeed, blockHash, blockNum, nil
+}
+
+// getBlockData parses the block-related data from the JSON input
+func getBlockData(input models.RunInput) (
+	blockHash *common.Hash, blockNum *uint64, err error) {
+	hashBytes, err := extractHex(input, "blockHash")
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "bad blockHash for vrf task")
```

Comment:
Same here, could re-use the existing `Block` struct

Reply:
Not sure which `Block` struct you have in mind, here, but I added a local `block` struct to pass these values around.